WeatherMonitoringSystem/Domain/Core/WeatherStation.cs

```diff
@@ -0,0 +1,35 @@
+﻿using WeatherMonitoringSystem.Bots;
+
+namespace WeatherMonitoringSystem.Core
+{
+    public class WeatherStation
+    {
+        private List<IWeatherBot> _bots = new();
+        private WeatherData _currentWeather;
+
+        public void AddBot(IWeatherBot bot)
+        {
+            _bots.Add(bot);
+        }
+
+        public void RemoveBot(IWeatherBot bot)
+        {
+            _bots.Remove(bot);
+        }
+
+        public void SetWeatherData(WeatherData data)
+        {
+            _currentWeather = data;
+            NotifyBots();
```

Comment:
why are we notifying the bots here? I noticed that you also do `bot.Update` again in the `UserInteraction`